pkg/v1/tkg/tkgctl/create_cluster.go

```diff
@@ -153,43 +153,51 @@ func (t *tkgctl) processManagementClusterInputFile(ir *InitRegionOptions) (bool,
-	isInputFileClusterClassBased := false
-	if t.tkgClient.IsFeatureActivated(config.FeatureFlagPackageBasedLCM) {
-		isInputFileClusterClassBased, clusterobj, err = t.checkIfInputFileIsClusterClassBased(cc.ClusterConfigFile)
-		if err != nil {
-			return isInputFileClusterClassBased, err
-		}
-		if isInputFileClusterClassBased {
-			if isTKGSCluster {
-				t.TKGConfigReaderWriter().Set(constants.ConfigVariableClusterName, clusterobj.GetName())
-				t.TKGConfigReaderWriter().Set(constants.ConfigVariableNamespace, clusterobj.GetNamespace())
-			} else {
-				err = t.processClusterObjectForConfigurationVariables(clusterobj)
-				if err != nil {
-					return isInputFileClusterClassBased, err
-				}
+	isInputFileClusterClassBased, clusterobj, err := t.checkIfInputFileIsClusterClassBased(cc.ClusterConfigFile)
+	if err != nil {
+		return isInputFileClusterClassBased, err
+	}
+	if isInputFileClusterClassBased {
+		if isTKGSCluster {
+			t.TKGConfigReaderWriter().Set(constants.ConfigVariableClusterName, clusterobj.GetName())
+			t.TKGConfigReaderWriter().Set(constants.ConfigVariableNamespace, clusterobj.GetNamespace())
+		} else {
+			err = t.processClusterObjectForConfigurationVariables(clusterobj)
+			if err != nil {
+				return isInputFileClusterClassBased, err
 			}
-			t.overrideClusterOptionsWithLatestEnvironmentConfigurationValues(cc)
 		}
+		t.overrideClusterOptionsWithLatestEnvironmentConfigurationValues(cc)
 	}
-	err = t.validateTKGSClusterClassFeatureGate(isInputFileClusterClassBased, isTKGSCluster)
+	if isInputFileClusterClassBased && !t.tkgClient.IsFeatureActivated(config.FeatureFlagPackageBasedLCM) {
+		return isInputFileClusterClassBased, fmt.Errorf(constants.ErrorMsgCClassInputFeatureFlagDisabled, config.FeatureFlagPackageBasedLCM)
+	}
+	err = t.validateTKGSFeatureGateStatus(isInputFileClusterClassBased, isTKGSCluster)
 	if err != nil {
 		return isInputFileClusterClassBased, err
 	}
 	return isInputFileClusterClassBased, nil
 }
 
-// validateTKGSClusterClassFeatureGate validates the TKGS clusterclass feature gate state
-func (t *tkgctl) validateTKGSClusterClassFeatureGate(isInputFileClusterClassBased, isTKGSCluster bool) error {
-	if t.tkgClient.IsFeatureActivated(config.FeatureFlagPackageBasedLCM) && isInputFileClusterClassBased && isTKGSCluster {
-		isFeatureActivated, err := t.featureGateHelper.FeatureActivatedInNamespace(context.Background(), constants.CCFeature, constants.TKGSClusterClassNamespace)
-		if err != nil {
-			return errors.Wrap(err, fmt.Sprintf(constants.ErrorMsgFeatureGateStatus, constants.CCFeature, constants.TKGSClusterClassNamespace))
-		}
-		if !isFeatureActivated {
-			return fmt.Errorf(constants.ErrorMsgFeatureGateNotActivated, constants.CCFeature, constants.TKGSClusterClassNamespace)
+// validateTKGSFeatureGateStatus validates the TKGS featuregate status for a specific feature
+func (t *tkgctl) validateTKGSFeatureGateStatus(isInputFileClusterClassBased, isTKGSCluster bool) error {
+	if isTKGSCluster {
```

Comment:
I think, We should not pass `isTKGSCluster` to this function and external invocation should handle this scenario.

Reply:
agree, done changes.